### ISSUE TITLE
[DS918+] iSCSI 관련 이슈입니다.

### ISSUE THREAD

**Issue by yymin1022** [PeterSuh-Q3/tinycore-redpill#30]
서버포럼에도 문의글을 올렸었고, 동일증상 문의를 올리신 분이 한분 더 계시긴 합니다만, 혹시나 싶어 이슈 남겨봅니다.
[제가 작성한 게시글](https://svrforum.com/nas/1217453?page=4)
[다른분의 동일증상 게시글](https://svrforum.com/nas/1231323?page=3)

장비는 오드로이드 H2 (J4105 / RTL8111G Dual) 입니다.
현재 최신 TCRP로 부트로더가 구성되어있고, DS918+ 7.2.1 DSM을 설치해두었습니다.

11월 TCRP 버전으로 구성하여 사용하던 중, 1월 초에 부트로더 업데이트를 위해 재부팅을 실시했고,
이후 부팅이 되지 않는 문제가 발생하였습니다.
그래서 부트로더와 HDD전체를 포맷하고 재설치 후 복원하여 사용하고 있는데요,
이 시점부터 iSCSI의 사용이 불가한 문제가 있습니다.

<img width="1186" alt="스크린샷 2024-02-24 15 20 16" src="https://github.com/PeterSuh-Q3/tinycore-redpill/assets/12806229/0458248e-79ca-4f48-86d5-d0e850b26525">

SAN Manager 상에서는 정상이라고 나오나, 외부에서 NAS 장비에 포트스캔을 해보면 3260포트 자체가 열려있지 않다고 감지되며, 실제로도 윈도우/리눅스/ESXi 모두에서 iSCSI를 검색하지 못합니다.
SAN Manager에서 LUN과 ISCSI를 모두 삭제한 뒤 다시 셋업해보기도 하였고,
최근 [서버포럼 게시글](https://svrforum.com/nas/1238513)에 따라 SAN Manager 자체를 제거한 뒤 재설치해보기도 하였으나, 증상이 동일합니다.

부트로더 자체를 재설치해보아도 문제가 동일한걸 보면, 현재 TCRP 버전에 문제가 있는건가 싶은데,
여기저기 로그를 확인해보아도 별다른 분석이 되지를 않네요.

혹시 확인해볼 부분이나 도움드릴 수 있는 분석파트가 있을까요?

**Comment by PeterSuh-Q3** [comment]
당시 1월초에 MAC-SPOOF 를 강제 적용한 문제가 있던 버전을 설치 하셨던것 같고 당시 SAN MAMAGER (iSCSI) 를 
손상시키는 문제가 있었던 버전이 동작 되었던것 같습니다.
한번 손상된 SAN MAMAGER (iSCSI)는 현재로서는 복구가 불가능합니다.

저도 아직 iSCSI 에 대한 충분한 지식이 없어서
san manager 서비스만 간신히 기동해서 vmm 만 동작하도록 만든 상태입니다.
더 깊은 내용을 아직 모르기에 san manager 가 완벽하게 복구 동작되도록 구현을 못드리고 있는 실정입니다.

지금은 문제가 해소된 버전을 처음부터 한번더 클린설치를 하시던가
아니면 시놀로지 안에서의 설정 초기화 방법 밖에는 없는것 같습니다.

설정 초기화로 진행하신다면 데이터 파티션은 건드려지면 안됩니다.
이떄는 DSM 이 설치되었던 시스템 파티션만 초기화가 되어야 합니다.

**Comment by yymin1022** [comment]
@PeterSuh-Q3
그렇다면 지금 DSM이 설치되어있는 상태에서, TCRP로 부팅하여 부트로더 업데이트만 진행하는 것은 의미가 없고,
새로 USB와 HDD를 포맷하며 DSM 재설치 후 데이터를 복원하면 사용이 가능하다는 것으로 이해하면 될까요?

**Comment by PeterSuh-Q3** [comment]
네, 현재는 당시 문제를 일으켰던 MAC-SPOOF ADDON 은 선택적 설치방식으로 바뀌여서 기본선택되어 들어가지 않도록 되어 있습니다.
새로 셋팅하시면 관련 문제가 없어야 합니다.

**Comment by yymin1022** [comment]
@PeterSuh-Q3
답변 감사드립니다.
혹시 해당 Add-On이 선택 설치로 변경된 시점이 언제인지 알 수 있을까요?

**Comment by PeterSuh-Q3** [comment]
이 시점 입니다.
https://svrforum.com/nas/1201762

**Comment by yymin1022** [comment]
@PeterSuh-Q3 
확인 감사드립니다.
제가 DSM을 최종적으로 재설치한 날짜는 1월 28일 경인데, 2월 2일에 패치가 배포된 것이군요.
그럼 이번에 DSM을 클린설치하면 이슈가 해결될 것으로 예상되네요.
주중에 시도해보고, 해결되는대로 Issue Close 처리하겠습니다!

**Comment by yymin1022** [comment]
TCRP로 부팅해서 최신버전으로 업데이트 시킨 뒤, Grub 메뉴에서 DSM 재설치로 부팅해서 설정/파일 유지 옵션으로 DSM 자체만 재설치 했더니 해결되었습니다.
감사합니다!

**Comment by yymin1022** [comment]
@PeterSuh-Q3 
지난 번 DSM 재설치로 iSCSI 이슈가 해결되었으나, DSM 재시작 시 동일증상과 함께 iSCSI 사용이 불가합니다.
아예 HDD 자체를 초기화하고 전체 클린설치를 해도 동일하며, 최초에 DSM 설치를 동반한 부팅시에만 iSCSI가 사용 가능합니다.
이후에 한번이라도 재부팅을 하는 경우, DSM 재설치를 수행해야만 iSCSI가 다시 사용 가능합니다.
다만, 이전에 발생하던 SAN Manager 오류 알림은 따로 발생하지는 않고, 단지 iSCSI만 사용이 불가한 증상입니다.
